### PR TITLE
Add option for labeled samples

### PR DESCRIPTION
**Pull request by nbardy**
The dreambooth setup can be very useful for more general finetuning as show by the community with examples such as style fine tuning.

https://twitter.com/rainisto/status/1584112851446542336

I've been using an approach adding a label per image as form of regularization, it's especially useful outside of the instance class training where the dataset is more diverse

I've been able to get some very nice wave photos with this approach
https://twitter.com/NicholasBardy/status/1586065285265530880

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
cc @patil-suraj here

**Comment by CrazyBoyM** [comment]
Hi, I add a --use_txt_as_label for you.
And this PR is here: https://github.com/nbardy/diffusers/pull/1

**Comment by nbardy** [comment]
This also adds model checkpointing now.

**Comment by patrickvonplaten** [comment]
Gently ping again @patil-suraj

File: examples/dreambooth/train_dreambooth.py [hunk 1]
```diff
@@ -184,6 +190,12 @@ def parse_args(input_args=None):
             " *output_dir/runs/**CURRENT_DATETIME_HOSTNAME***."
         ),
     )
+    parser.add_argument(
```
Comment (patrickvonplaten) [inline examples/dreambooth/train_dreambooth.py:193]: This is not really related to the PR no? Could we maybe make this a seperate PR?

File: examples/dreambooth/train_dreambooth.py [hunk 2]
```diff
@@ -326,16 +360,43 @@ def get_full_repo_name(model_id: str, organization: Optional[str] = None, token:
         return f"{organization}/{model_id}"
 
 
+def save_model(accelerator, unet, text_encoder, args, step=None):
```
Comment (patrickvonplaten) [inline examples/dreambooth/train_dreambooth.py:363]: I have a hard time seeing how this is related to the PR of adding an option for labeled samples? Could we maybe try to keep PRs laser focused on one thing? :-)

**Comment by patrickvonplaten** [comment]
Regarding checkpointing @patil-suraj do we need this for dreambooth still? Think it might make sense to add it no?

File: examples/dreambooth/requirements.txt [hunk 3]
```diff
@@ -4,4 +4,5 @@ torchvision
 transformers>=4.21.0
 ftfy
 tensorboard
-modelcards
\ No newline at end of file
+modelcards
+faiss-gpu==1.7.2
```
Comment (patrickvonplaten) [inline examples/dreambooth/requirements.txt:8]: why do we need `faiss` now?
